File: backend/schemas/account/role.py
from typing import Optional
from pydantic import BaseModel
from typing import Optional, List
from core.security import Permission as SecurityPermission
from datetime import datetime

class RoleBase(BaseModel):
    """角色基础模型，定义角色的基本属性"""
    name: str  # 角色名称
    description: Optional[str] = None  # 角色描述，可选字段

class RoleCreate(RoleBase):
    """用于创建角色的模型，继承自 RoleBase"""
    permissions: List[str] = [SecurityPermission.STOCK_READ.value]  # 角色权限列表，默认包含STOCK-read权限


class RoleResponse(RoleBase):
    """用于返回角色信息的模型，继承自 RoleBase"""
    id: int
    create_time: datetime  # 创建时间
    update_time: datetime  # 修改时间
    
class RoleUpdate(RoleBase):
    """用于更新角色的模型，继承自 RoleBase"""
    pass
    
class RoleWithPermissions(RoleResponse):
    """包含权限信息的角色模型"""
    permissions: List[str] = []
    
class UpdateRolePermissions(BaseModel):
    """用于更新角色权限的模型"""
    permission_ids: List[str]

class PaginationResult(BaseModel):
    """通用分页结果模型"""
    total: int  # 总记录数
    page: int  # 当前页码
    page_size: int  # 每页记录数
    total_pages: int  # 总页数
    data: List[RoleWithPermissions]  # 分页数据列表

File: frontend/src/services/types/role.ts
// 角色相关类型
export interface RoleResponse {
  id: number;
  name: string;
  description: string;
}

export interface RoleWithPermissions {
  id: number;
  name: string;
  description: string;
  create_time: string;
  update_time: string;
  is_delete: boolean;
  permissions: string[];
}

export interface RoleCreate {
  name: string;
  description: string;
  permissions: string[];
}

export interface RoleUpdate {
  name: string;
  description: string;
}

export interface UpdateRolePermissions {
  permission_ids: string[];
}

// 注意：Permission 和 UserInfo 类型已在专门的类型文件中定义
// Permission -> @/services/types/permission
// UserInfo -> @/services/types/auth
// 为了避免类型冲突，这里不再重复定义

// 角色表单类型
export interface RoleForm {
  name: string;
  description: string;
  permissions: string[];
}